tools/topology/topology2/sof/sof-hda-generic.conf

```diff
@@ -0,0 +1,455 @@
+# include paths
+<searchdir:include>
+<searchdir:include/common>
+<searchdir:include/components>
+<searchdir:include/controls>
+<searchdir:include/dais>
+<searchdir:include/pipelines>
+
+# Include class definitions
+<vendor-token.conf>
+<tokens.conf>
+<volume-playback.conf>
+<volume-demux-playback.conf>
+<volume-capture.conf>
+<passthrough-capture.conf>
+<eq-iir-volume-capture.conf>
+<dai.conf>
+<virtual.conf>
+<data.conf>
+<pcm.conf>
+<pcm_caps.conf>
+<fe_dai.conf>
+<hda.conf>
+<hw_config.conf>
+<manifest.conf>
+<route.conf>
+<variables.conf>
+
+# Define variables. These can be overridden from command-line while building the conf file.
+Define {
+	DYNAMIC_PIPELINE	0
+	DMIC_CH_COUNT		0
+	EQIIR_BYTES		"highpass_40hz_0db_48khz"
```

Comment:
This would be good to prefix with endpoint name, to prepare for e.g. DMIC0, HSEAR, HSMIC. etc.